develop/custom-resources.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Custom Resources
+description: A comprehensive guide on how to use the resource system and load custom resources during the runtime.
+authors:
+  - Manchick
+---
+
+# Custom Resources {#what-are-custom-resources}
+
+Some resources in Minecraft are loaded during the runtime, instead of being hard-coded into the game.
+Each resource type falls into one of two categories: **Client Resources** and **Server Data**.
+Client Resources are the ones that are loaded from a **resource pack**, thus being fully client-side, and accessible at any time.
+Server Data, on the other hand, is used for various tasks on the server, and is loaded from a **data pack**.
+
+## Deciding on your Resource Type {#deciding-on-a-resource-type}
+
+Which type you'll use is completely up to you and your needs. Most of the time, the type depends on
+the environment you're working with. For example, for recipes for a custom block, you'd want to use the
+server data, while client resources might be handy if you want to allow the creation of new themes for your screen.
+
+# Creating a JsonDataLoader
+
+> This article assumes you have a class for the object you want to read with a proper deserialization method.
+> For example, `MyClass#deserialize(JsonElement element)`. If you're unsure of how to create such methods, consider
+> taking a closer look at what a [Codec](../develop/codecs.md) is.
+
+The `JsonDataLoader` will function as the main key point in our system. It provides the `apply(Map<Identifier, JsonElement> prepared, ResourceManager manager, Profiler profiler)`
+method, which we'll override to load the read files into our storage.
+
+> We'll create a `JsonDataLoader` for loading JSON files from the `fruit` folder
+> in a data pack, deserialize them into actual fruits, and store them
+> for further retrieval.
+
+Let's start by creating a class itself. It should extend `JsonDataLoader`, and override the constructor
+and the `apply` method.
+
+```java
+public class FruitDataLoader extends JsonDataLoader {
```

Comment:
Needs to be in the reference mod. Just because it uses mc classes which may change in the future (ResourceManager)

Reply:
Sure! I added a new class, which is just empty. I really didn't want to do it though, that's why I decided to use raw code in the previous version.